apm-sniffer/apm-sdk-plugin/grizzly-2.3.x-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/grizzly/v2/HttpServiceInterceptor.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.grizzly.v2;
+
+import java.lang.reflect.Method;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.ContextSnapshot;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+import org.glassfish.grizzly.http.server.Response;
+
+public class HttpServiceInterceptor implements InstanceMethodsAroundInterceptor {
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        Object[] context = (Object[]) objInst.getSkyWalkingDynamicField();
+        ContextSnapshot contextSnapshot = (ContextSnapshot) context[1];
+        AbstractSpan span = ContextManager.createLocalSpan("GrizzlyRunService");
+        span.setComponent(ComponentsDefine.GRIZZLY);
+        ContextManager.continued(contextSnapshot);
+
+    }
+
+    @Override
+    public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, Object ret) throws Throwable {
+        Object[] context = (Object[]) objInst.getSkyWalkingDynamicField();
+        AbstractSpan abstractSpan = (AbstractSpan) context[0];
```

Comment:
Is the target class created for every request? Otherwise, you will face concurrency issue, as the dynamic field is shared.
The MethodInvocationContext is the only safe place to share between before and after for every method.

Reply:
get it, i will put it to request scope